packages/sanity/src/core/form/studio/tree-editing/components/TreeEditingMenu.tsx

```diff
@@ -115,13 +116,18 @@ function MenuItem(props: TreeEditingMenuItemProps) {
   const selected = useMemo(() => isEqual(item.path, selectedPath), [item.path, selectedPath])
   const isArrayParent = useMemo(() => !isArrayItemPath(item.path), [item.path])
 
+  const {value} = useValuePreview({
```

Comment:
Small nit: Since this code seems to be repeated across, maybe move it to a utils? 🤔